### PR TITLE
Add CI/CD pipeline for linting

### LINKED ISSUE #1: Add CI for linting, type-checking
We should have a CI pipeline like we do in the Next.js example:

https://github.com/upleveled/next-js-example-spring-2023-vienna-austria/blob/main/.github/workflows/test-playwright-and-deploy-to-fly-io.yml

### PR DESCRIPTION
**Pull request by Eprince-hub**
Closes https://github.com/upleveled/external-apis-example-spring-2023-vienna-austria/issues/1

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -19,7 +19,7 @@
   "devDependencies": {
     "@types/react": "^18.2.8",
     "@types/react-dom": "^18.2.4",
-    "eslint-config-upleveled": "^4.5.0",
+    "eslint-config-upleveled": "^5.0.0",
```
Comment (Eprince-hub) [inline package.json:22]: Update UpLeveled ESLint configuration to follow the latest best practices

File: app/api/sessions/route.ts [hunk 2]
```diff
@@ -1,7 +1,14 @@
...
 import { stripeClient } from '../../../utils/stripe';
 
-export async function POST(request: NextRequest) {
+type StripeResponseBody = {
+  sessions: Stripe.Checkout.Session;
+};
+
+export async function POST(
+  request: NextRequest,
+): Promise<NextResponse<StripeResponseBody>> {
```
Comment (Eprince-hub) [inline app/api/sessions/route.ts:11]: Import the correct type from Stripe to use as the response body for the nest.js route handler and also rename the file from `route.tsx` to `route.ts`, I guess the `tsx` extension was not intended

File: app/api/route.ts [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+
+type StripeResponse = {
+  sessions: string;
+};
+
+export async function GET(): Promise<NextResponse<StripeResponse>> {
+  return await NextResponse.json({
+    sessions: '/api/sessions',
+  });
+}
```
Comment (Eprince-hub) [inline app/api/route.ts:11]: Add the `StripeResponse` type to route handler

File: .github/workflows/lint-check-types.yml [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+          node-version: '18'
+          cache: 'pnpm'
+      - name: Install dependencies
+        run: pnpm install
+      - name: Run TypeScript Compiler
+        run: pnpm tsc
+      - name: Run ESLint
+        run: pnpm eslint . --max-warnings 0
+      - name: Run linter
+        run: pnpm lint
```
Comment (Eprince-hub) [inline .github/workflows/lint-check-types.yml:28]: Linting and type-checking setup, the setup doesn't need Postgres or migrations setup as its not part of this repo

**Comment by karlhorky** [comment]
Implemented in `external-apis-example-fall-2023-atvie`:

- https://github.com/upleveled/external-apis-example-fall-2023-atvie/commit/2cb73f9a58acf23a4db4b740a96af8181448f0f3
- https://github.com/upleveled/external-apis-example-fall-2023-atvie/commit/4eba89d37159eb117b9f07aa17a5aa33889072b3
- https://github.com/upleveled/external-apis-example-fall-2023-atvie/commit/1e13037ab3028f202b6fa12480112520d6f69e99

**Comment by Eprince-hub** [comment]
Closing because the example has been implemented in the API lectures as mentioned in [the comment above](https://github.com/upleveled/external-apis-example-spring-2023-atvie/pull/2#issuecomment-1759445436) by @karlhorky

**Pull request closed without merging**